pkg/pull/pull.go

```diff
@@ -351,14 +358,24 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 				}
 			}
 
-			for _, newChart := range newHelmCharts {
+			for _, newChart := range v1Beta1HelmCharts {
 				if prevChart.Spec.Chart.Name != newChart.Spec.Chart.Name {
 					continue
 				}
 				if prevChart.Spec.UseHelmInstall != newChart.Spec.UseHelmInstall {
 					return "", errors.Errorf("deployment method for chart %s has changed", newChart.Spec.Chart.Name)
 				}
 			}
+
+			for _, newChart := range v1Beta2HelmCharts {
+				if prevChart.Spec.Chart.Name != newChart.Spec.Chart.Name {
```

Comment:
should this be release name instead of chart name?

Reply:
I would agree.  The v1beta1 check uses chart name, but that could be a bug.  I can change both.